### PR TITLE
Add support for scrollable tabs

### PR DESCRIPTION
**Pull request by StefKors**
Moving https://github.com/product-os/jellyfish/pull/3737 over to Rendition

Change-type: minor
Signed-off-by: Graham McCulloch <graham@balena.io>

Connects-to: # 
See: <url> 
Depends-on: https://github.com/balena-io-modules/rendition/pull/1120
Change-type: major|minor|patch 

---
##### Contributor checklist

- [x] I have rebuilt the README with `npm run build:docs`
- [x] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [x] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by StefKors** [comment]
@grahammcculloch I've moved the CSS to the component instead of the theme file because it seems to line-up more with the other rendition components like that. I'm happy to move it back if you prefer it in the `theme` file.

I've also added support for CSS scroll snapping as a nice interaction improvement for mobile users.

**Comment by grahammcculloch** [comment]
@StefKors think you need to re-generate the screenshots again for the tests to pass? Once that's done and the PR is out of draft I'll be happy to approve it!

**Comment by StefKors** [comment]
Depends-on: https://github.com/balena-io-modules/rendition/pull/1120

**Comment by StefKors** [comment]
@grahammcculloch could you re-review?

**Comment by StefKors** [comment]
Converting to draft until I add `<` `>` arrows

**Comment by StefKors** [comment]
I'm picking this PR up again. @sradevski After trying out some options to add navigation buttons to the grommet tab component, I'm running into two problems
1. I would need to resort to some easy to break CSS tricks to absolutely position the buttons and add margins + paddings in the correct places
2. To implement the scroll to tab behaviour it would require using react refs. Getting these to the correct spot will also be very messy.

**I want to propose moving forward and keeping the scroll bars for now.** If we still want to have buttons and no scrollbars, I can start a separate PR to first replace the grommetTabs component, with a DIY one or a tabs component from a different component lib that already supports buttons.

@sradevski let me know what you think

File: src/components/Tabs/story.js [hunk 1]
```diff
@@ -36,3 +36,106 @@ storiesOf('Next/Tabs', module)
...
+          <Tab title='This is a very long tab name'>
+            <Txt mt={3}>Here is tab #1</Txt>
+          </Tab>
+          <Tab title='This is another very long tab name'>
+            <Txt mt={3}>Here is tab #2</Txt>
+          </Tab>
+          <Tab title='And this is a third very long tab name testing'>
+            <div>
+              <Txt mt={3}>
+                Hereeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee is tab #3
```
Comment (sradevski) [inline src/components/Tabs/story.js:61]: Let's replace this with Lorem Ipsum as a more esthetic option :)

File: src/components/Tabs/index.tsx [hunk 2]
```diff
@@ -1,10 +1,88 @@
...
+			`;
+		}
+	}}
+`;
+
+export const Tab = ({ compact, title, ...props }: InnerTabProps) => {
+	return (
+		<Flex>
+			<TabTitle compact={compact} title={title} {...props}>
+				{/* {title} */}
```
Comment (sradevski) [inline src/components/Tabs/index.tsx:65]: I don't think this is intended

File: src/components/Tabs/index.tsx [hunk 3]
```diff
@@ -1,10 +1,88 @@
...
+import React from 'react';
 import asRendition from '../../asRendition';
+import { Flex } from '../Flex';
 
-const TabsBase = ({ ...props }: TabsProps) => {
-	return <GrommetTabs justify="start" {...props} />;
+interface InnerTabProps extends TabProps {
+	compact?: boolean;
+	index: number;
+	length?: any;
```
Comment (sradevski) [inline src/components/Tabs/index.tsx:16]:
```suggestion
	length?: number;
```

**Comment by sradevski** [comment]
@StefKors the only problem I have with scroll-only option is that it is not clear when there are tabs that are hidden. We can potentially add some blur effect when there is an overflow at the end of he tab container, but I don't think that is any simpler. I'd be fine with it as it is TBH just so we can get the ball rolling, but it is not great UX.

File: src/components/Tabs/story.js (hunk 1 above)
Comment (StefKors) [reply src/components/Tabs/story.js:61]: agree :+1:

File: src/components/Tabs/index.tsx (hunk 2 above)
Comment (StefKors) [reply src/components/Tabs/index.tsx:65]: removed :+1:

File: src/components/Tabs/index.tsx (hunk 3 above)
Comment (StefKors) [reply src/components/Tabs/index.tsx:16]: fixed :+1:

**Comment by StefKors** [comment]
@sradevski in this case the scrollbar will show that the tabs are hidden. Shall we move forward with the scroll bar for now, and discuss a better UX using either buttons, blurs or something else at a later date?

**Comment by sradevski** [comment]
@StefKors yeah sure, otherwise we might never get this done :)

**Review by sradevski — APPROVED** [review]

**Comment by sradevski** [comment]
@StefKors @grahammcculloch requested changes so he'll have to approve it

**Review by grahammcculloch — APPROVED** [review]
:shipit:

**Comment by grahammcculloch** [comment]
@balena-ci rebase

**Pull request merged**